org.OpenT2T.Sample.SuperPopular.Thermostat/Wink Thermostat/v2/thingTranslator.js

```diff
@@ -0,0 +1,117 @@
+"use strict";
+
+var q = require('q');
+var wh = require("opent2t-translator-helper-wink");
+
+// This code uses ES2015 syntax that requires at least Node.js v4.
+// For Node.js ES2015 support details, reference http://node.green/
+
+const EventEmitter = require("events");
+
+function validateArgumentType(arg, argName, expectedType) {
+    if (typeof arg === 'undefined') {
+        throw new Error('Missing argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + '.');
+    } else if (typeof arg !== expectedType) {
+        throw new Error('Invalid argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + ', got: ' + (typeof arg));
+    }
+}
+
+/**
+ * This translator class implements the "InterfaceA" interface.
+ * Since the interface includes a signal (notifiable property), this class
+ * extends from the node's built-in event-emitter class.
+ */
+class WinkThermostat extends EventEmitter {
+
+    constructor(deviceInfo) {
+        super(); // Construct EventEmitter base
+        this.initDevice(deviceInfo);
+    }
+
+    initDevice(deviceProps) {
+        console.log('Initializing device.');
```

Comment:
There is no point in trying to retain compatibility here with the existing bridge code. It already has no chance of working because of how the module exports are done differently: the class is exported rather than an object instance.

The bridge will soon (in a few weeks I guess) consume the OpenT2T client. Through the DeviceAccessor APIs projected through the client, it will have no trouble calling the constructor.

---
In reply to: [73779700](https://github.com/openT2T/translators/pull/37#discussion_r73779700) [](ancestors = 73779700,73779241)